apps/emqx/include/emqx_api_lib.hrl

```diff
@@ -0,0 +1,36 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2021-2023 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-ifndef(EMQX_API_LIB_HRL).
+-define(EMQX_API_LIB_HRL, true).
+
+-define(ERROR_MSG(CODE, REASON), #{code => CODE, message => emqx_misc:readable_error_msg(REASON)}).
```

Comment:
(Maybe for future discussion): I always wondered why we should have all the responses as macros. We do not match against them anywhere, and the performance impact should be neglectable 🤔

Reply:
I think it vastly improves readability and consistency, my goal is rather to ONLY use those macros throughout rather than constructing all those responses by hand.

Or is it, that you would favor function calls instead?

Reply:
(Also I match against them in my tests)

Reply:
>that you would favor function calls instead?
Yes, I mean 
```
error_msg(Code, Reason) ->
...
```
vs
```
-define(ERROR_MSG(CODE, REASON), ...).
```